hoomd/snapshot.py

```diff
@@ -109,3 +109,38 @@ def replicate(self, nx, ny, nz):
 
     def _broadcast_box(self):
         self._cpp_obj._broadcast_box(self._comm.cpp_mpi_conf)
+
+    @classmethod
+    def _from_gsd_snapshot(cls, gsd_snapshot):
+        snap = cls()
+        # Set all particle attributes in snap from gsd_snapshot
+        for key in vars(gsd_snapshot.particles):
+            val = vars(gsd_snapshot.particles)[key]
+            if val is not None:
+                try:
+                    setattr(snap.particles, key, val)
+                except AttributeError:
```

Comment:
I'd prefer to see more explicit code here rather than exception tests which could be fragile. For example, set `type` and `N` explicitly and then loop over the known numpy arrays that need to be copied with `[:]`.

Reply:
yeah, I thought I was being clever, but you're right that it's less error-prone (and easier to follow) if I just set each attribute manually.